godot-macros/src/class/godot_api.rs

```diff
@@ -32,8 +52,17 @@ pub fn attribute_godot_api(input_decl: venial::Item) -> ParseResult<TokenStream>
     };
 
     if decl.trait_ty.is_some() {
+        if meta.to_string() != "" {
+            return bail!(
+                meta,
+                "#[godot_api] on a trait implementation currently does not support any parameters"
+            );
+        }
```

Comment:
I'm confused what this is supposed to catch

Reply:
A user could do ``#[godot_api(a, b, x=y)]``, which in the old implementation would just be ignored, and now would throw an error.

The main thought behind this was to catch people accidentally adding ``secondary`` to a trait impl block instead of an inherent impl block.

Reply:
Very good corner case -- could you add this explanation to the code itself as a comment?

Reply:
```rust
    if decl.trait_ty.is_some() {
        // 'meta' contains the parameters to the macro, that is, for `#[godot_api(a, b, x=y)]`, anything inside the braces.
        // We currently don't accept any parameters for a trait `impl`, so show an error to the user if they added something there.
        if meta.to_string() != "" {
```

Reply:
Thanks a lot!